### ISSUE TITLE
[Natural Disasters DLC] Timing between announcement and event too short

### ISSUE THREAD

**Issue by mitgobla** [dymanoid/RealTime#246]
Natural Disasters DLC adds random disaster events. These events get announced on the radio when they are going to approach the city **in a few days**. Then, when the disaster is striking/has striked the city, it gets announced again.

However, with Real Time enabled, these events happen within meer hours of each other (the announcement -> to the actual disaster) meaning cims have literally no time to evacuate before the disaster happens. 

It appears Real Time does not increase the time between announcement of the disaster to the actual disaster event.

**Comment by dymanoid** [comment]
Yes, the behavior you describe indeed applies. However, Real Time doesn't change anything here. In the vanilla game, the citizens have exact the same "time" for evacuation - "time" in sense of "distance to shelter by citizens movement velocity". Real Time doesn't change how fast citizens (or vehicles) move, neither changes it the progress of the disasters.

Of course, "a few days" in the announcement is now wrong. Probably, I could change the disasters in that way. But isn't it not too boring then? A few game days with Real Time mean several hours of real time - the players might think then that the game just "forgets" about the disaster (like, there's a bug in the game caused by the mod). Then, the disaster strikes "unexpectedly" a few game days later.

What do you think?

Also tagging @aubergine10 for valuable feedback.

**Comment by aubergine10** [comment]
I also experience the issue described in OP. It's one of the reasons I have disabled some of the more devastating disasters, because there was insufficient time to respond to them (eg. the evacuation bus would only be half way through its route before disaster struck).

Thunderstorms are the best example I can think of. With no detection buildings, you only get short warning "a few hours". However, you can build a **Weather Radar** to increase forecast range up to a maximum of **14 days warning**. With RT, that doesn't seem to happen so it makes the detection buildings somewhat useless.

The list of all detection buildings:

* Deep Space Radar - meteorites
* Earthquake Sensor
* Tsunami Warning Buoy
* Firewatch Tower - forest fires (and general fires)

Note also that these disaster detection buildings are just assets, and therefore there are alternatives with different stats in the Steam Workshop. For example, the [Geological Survey Office](https://steamcommunity.com/sharedfiles/filedetails/?id=822673042) is like a giant earthquake sensor that covers the entire map. Whether the 'forecast period' is just a stat on the building or hard-coded in to an AI class somewhere, I have no idea, so that would need some investigations.

BTW, there are some mods that potentially further compound the issue and may also mess with RT's disaster/weather choices (and subsequent cim schedules - eg. fleeing from rain):

* [Natural Disasters Overhaul](https://steamcommunity.com/sharedfiles/filedetails/?id=1801953480) - runs it's own schedule, so this will definitely cause issues with RT.
* [Ragnarok](https://steamcommunity.com/sharedfiles/filedetails/?id=811352708) - never actually tried this one, but flagging it up as it obviously does change _something_ relating to disasters (whether that's just magnitude or something more elaborate I don't know; haven't investigated that mod much).
* [Rainfall](https://steamcommunity.com/sharedfiles/filedetails/?id=698395457) - Not sure if this causes too much problem, but it has "non-vanilla storms" and from mod description it seems it has it's own variant of the forecast system, complete with chirps. I suspect it will at least change duration of storms, and would thus mismatch with RT's cim schedules, but haven't verified that.

Related: https://github.com/dymanoid/CitiesSkylinesModWishList/issues/1 (which I will be adding to shortly :) )

**Comment by mitgobla** [comment]
As aubergine10 mentioned, there is almost no time for citizens to prepare for a disaster. So a fix would mean citizens would have time to make use of the shelters, and it would make the detection buildings useful. 

You mentioned that the disaster might strike unexpectedly, however the DLC does add an information ticker at the top left when a disaster will be approaching in the upcoming days. That would then give you enough time to call an evacuation.

I do agree though that having a disaster take a few real life hours to arrive would get boring, so rather than having disaster events happening 14 in-game days later, it could be shortened.

**Comment by aubergine10** [comment]
Just done some quick testing of the various detection buildings.

There's really only weather radar that gives long-term warning; the others might give a bit of extra warning, but their primarily focus on identifying where the disaster will strike.

So, even in base game there's quite a bit of time for storm/tornado warnings - as there would be IRL, I guess. For Real Time, I think maybe the storm warnings could be changed from max 14 days to max 14 hours - so you still get some realism, but for gameplay you're not left waiting too long.
